scripts/linters/general_purpose_linter_test.py

```diff
@@ -581,14 +581,35 @@ def mock_readlines(unused_self, unused_filepath):
 
         with readlines_swap:
             linter = general_purpose_linter.GeneralPurposeLinter(
-                [INVALID_DEV_MODE_IN_CONSTANT_FILEPATH], FILE_CACHE)
+                [CONSTANT_FILEPATH], FILE_CACHE)
             lint_task_report = linter.check_bad_patterns()
         self.assert_same_list_elements([
             'Please set the DEV_MODE variable in constants.ts'
             'to true before committing.'], lint_task_report.trimmed_messages)
         self.assertEqual('Bad pattern', lint_task_report.name)
         self.assertTrue(lint_task_report.failed)
 
+    def test_invalid_emulator_mode_in_constant_ts(self):
+        def mock_readlines(unused_self, unused_filepath):
+            return (
+                'Copyright 2020 The Oppia Authors. All Rights Reserved.',
+                ' * @fileoverview Initializes constants for '
+                'the Oppia codebase.',
+                '"EMULATOR_MODE": false\n',)
+
+        readlines_swap = self.swap(
+            pre_commit_linter.FileCache, 'readlines', mock_readlines)
+
+        with readlines_swap:
+            linter = general_purpose_linter.GeneralPurposeLinter(
+                [CONSTANT_FILEPATH], FILE_CACHE)
+            lint_task_report = linter.check_bad_patterns()
+        self.assert_same_list_elements([
+            'Please set the EMULATOR_MODE variable in constants.ts'
+            'to true before committing.'], lint_task_report.trimmed_messages)
```

Comment:
Can you please assert the entire message? (We are planning to remove assert_same_list_elements function.)

Reply:
Done. Also did a refactor of the constants patters since they were broken.